app/questionnaire/router.py

```diff
@@ -224,22 +224,26 @@ def _get_return_to_location_url(
         if not return_to:
             return None
 
-        if return_to == "calculated-summary" and self.can_access_location(
-            Location(
-                block_id=return_to_block_id,
-                section_id=location.section_id,
-                list_item_id=location.list_item_id,
-            ),
-            routing_path,
-        ):
-            return url_for(
-                "questionnaire.block",
-                block_id=return_to_block_id,
-                list_name=location.list_name,
-                list_item_id=location.list_item_id,
-                return_to=return_to,
-                _anchor=return_to_answer_id,
-            )
+        if return_to == "grand-calculated-summary":
+            if url := self._get_return_to_for_grand_calculated_summary(
```

Comment:
Once the progress source merges, worth updating a schema to use that. We need to check what happens if grand summary become inaccessible.
On another note, the nested if can be both single line statements?. Applies to the one below also.

```py
if return_to == "something" and url := get_url():
  return url
```

Reply:
Its currently possible to test what happens when it becomes inaccessible in the `cross_section_dependencies` schema, the grand calculated summary depends on providing an answer to part b of question 3 - that said, I haven't made that clear, so I'll update that schema to make it better. I'll add a functional test that checks routing for this case and once progress is in it can just be updated with a slightly different when rule

fixed if statements